src/query/plan/rule_based_planner.hpp

```diff
@@ -232,6 +252,10 @@ class RuleBasedPlanner {
           throw utils::NotYetImplemented("clause '{}' conversion to operator(s)", clause->GetTypeInfo().name);
         }
       }
+      // Is this the only situation that should be covered
+      if (!handle_empty_result && !query_part.remaining_clauses.empty()) {
+        input_op = std::make_unique<EmptyResult>(std::move(input_op));
+      }
```

Comment:
It is a bit strange that if `handle_empty_result` is false that we want to create `EmptyResult`, maybe rename the bool, or something just to make it more clear